src/sdk/pynni/nni/analysis_utils/sensitivity/torch/sensitivity_analysis.py

```diff
@@ -0,0 +1,257 @@
+
+    def analysis(self, val_args=None, val_kwargs=None, start=0, end=None):
+        """
+        This function analyze the sensitivity to pruning for
+        each conv layer in the target model.
+        If %start and %end are not set, we analyze all the conv
+        layers by default. Users can specify several layers to
+        analyze or parallelize the analysis process easily through
+        the %start and %end parameter.
+
+        Parameters
+        ----------
+            start:
+                Layer index of the sensitivity analysis start
+            end:
+                Layer index of the sensitivity analysis end
+            val_args:
+                args for the val_function
+            val_kwargs:
+                kwargs for the val_funtion
+                The val_funtion will be called as:
+                    val_function(*val_args, **val_kwargs)
+
+        Returns
+        -------
+            sensitivities:
+                dict object that stores the trajectory of the
+                accuracy when the prune ratio changes
+        """
+        if not end:
+            end = self.layers_count
+        assert start >= 0 and end <= self.layers_count
+        assert start <= end
+        if val_args is None:
+            val_args = []
+        if val_kwargs is None:
+            val_kwargs = {}
+        # Get the validation accuracy before pruning
+        if self.ori_acc is None:
+            self.ori_acc = self.val_func(*val_args, **val_kwargs)
+        namelist = list(self.target_layer.keys())
+        for layerid in range(start, end):
+            name = namelist[layerid]
+            self.sensitivities[name] = {}
+            for sparsity in self.sparsities:
+                # Calculate the actual prune ratio based on the already pruned ratio
+                sparsity = (
+                    1.0 - self.already_pruned[name]) * sparsity + self.already_pruned[name]
+                # TODO In current L1/L2 Filter Pruner, the 'op_types' is still necessary
+                # I think the L1/L2 Pruner should specify the op_types automaticlly
+                # according to the op_names
+                cfg = [{'sparsity': sparsity, 'op_names': [
+                    name], 'op_types': ['Conv2d']}]
+                pruner = self.Pruner(self.model, cfg)
+                pruner.compress()
+                val_acc = self.val_func(*val_args, **val_kwargs)
+                logger.info('Layer: %s Sparsity: %.2f Accuracy: %.4f',
+                            name, sparsity, val_acc)
+
+                self.sensitivities[name][sparsity] = val_acc
+                pruner._unwrap_model()
+                del pruner
+                # if the accuracy drop already reach the 'early_stop'
+                if val_acc + self.early_stop < self.ori_acc:
+                    break
+
+            # reset the weights pruned by the pruner, because
+            # out sparsities is sorted, so we donnot need to reset
```

Comment:
"out" sparsities?

Reply:
it's a typo. "out" => "input"
what I want to say is "the input sparsity list is sorted."
I'll fix that. Thanks~